src/rust/ide/src/model/synchronized/execution_context.rs

```diff
@@ -219,4 +246,34 @@ mod test {
             assert!(context.pop().await.is_err());
         })
     }
+
+    #[test]
+    fn attaching_visualizations() {
+        let exe_id   = model::execution_context::Id::new_v4();
+        let path     = controller::module::Path::from_module_name("Test");
+        let root_def = DefinitionName::new_plain("main");
+        let model    = model::ExecutionContext::new(root_def);
+        let ls       = language_server::MockClient::default();
+        let vis      = Visualization {
+            id: model::execution_context::VisualizationId::new_v4(),
```

Comment:
the name `new_v4` is enigmatic. What does it do? Is it our API? If its our, thisshould be changed. If its not, we should have wrapper around that. May be not part of this PR, but then add task for it pls.

Reply:
This pattern is used on all IDs, since they are just type aliases instead of newtypes.